gitpod-io/gitpod
# PVC integration tests: TestBackup and TestRegularWorkspaceTasks and TestWorkspaceEnableAndDisablePVC

## Description

Add PVC feature flag to TestBackup
- Create a fresh new workspace with Git initializer
- Write a `foobar.txt` file
- Stop the workspace
- Recreate the workspace
- Make sure the `foobar.txt` file exists

Add PVC feature flag to TestRegularWorkspaceTasks
- Create a fresh new workspace with Git initializer and Gitpod tasks
- Make sure each task generated file exists

Add PVC integration test to TestWorkspaceEnableAndDisablePVC
- Open a fresh new workspace
- Write a `foobar.txt` file
- Stop the workspace
- Relaunch the workspace with the PVC feature flag
- Make sure the `foobar.txt` file exists
- Stop the workspace
- Relaunch the workspace without the PVC feature flag
- Make sure the `foobar.txt` file exists

## Related Issue(s)

#12497

## How to test

```console
cd test

go test  -run ^TestBackup$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v

go test  -run ^TestRegularWorkspaceTasks$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v

go test  -run ^TestWorkspaceEnableAndDisablePVC$ github.com/gitpod-io/gitpod/test/tests/components/ws-manager -count=1 -namespace=default -kubeconfig=/home/gitpod/.kube/config -v
```

## Release Notes

```release-note
None
```

## Documentation


## Werft options:
